src/sagemaker/model_monitor/model_monitoring.py

```diff
@@ -2006,6 +2013,42 @@ def create_monitoring_schedule(
             logger.error(message)
             raise ValueError(message)
 
+        # error checking and validation logic for dashboard name
+        if (not enable_cloudwatch_metrics) and enable_automatic_dashboard:
+            message = "Could not create automatic dashboard. Please set enable_cloudwatch_metrics to True."
+            logger.error(message)
+            raise ValueError(message)
+
+        if enable_automatic_dashboard:
+            # verify that the provided dashboard name is not taken
+            dashboard_name = monitor_schedule_name if dashboard_name is None else dashboard_name
+
+            dashboard_name_validation = bool(re.match(r"^[0-9A-Za-z\-_]{1,255}$", dashboard_name))
+            if dashboard_name_validation == False:
+                message = (
+                    f"Dashboard name {dashboard_name} is not a valid dashboard name. "
+                    "Dashboard name can be at most 255 characters long "
+                    "and valid characters in dashboard names include '0-9A-Za-z-_'."
+                )
+                logger.error(message)
+                raise ValueError(message)
+
+            try:
```

Comment:
Why do we need a try except expression here?

Reply:
We need it since if the dashboard exists we don't want to overwrite it.